### ISSUE TITLE
CheckExists and CheckExists

### ISSUE THREAD

**Issue by GuyMicciche** [PCLExt/PCLExt.FileStorage#21]
```
var rootFolder = new LocalRootFolder();
ExistenceCheckResult exists = await rootFolder.CheckExistsAsync("test.zip");
```

This like causes an exception in UWP and stops my app. Also,

```
var rootFolder = new LocalRootFolder();
ExistenceCheckResult exists = rootFolder.CheckExists("test.zip");
```
This causes `Exception thrown: 'System.IO.FileNotFoundException' in System.Private.CoreLib.dll ` and freezes the app.

Am I using these incorrectly, or is there an issue?? Please help. I need to check if files exists. Thank you so much for your help. Seems like a great extension.

**Comment by Aragas** [comment]
Unfortunately, I didn't had much experience with UWP and it's permission system, so I can't help with it right now. I'll temporarily remove the UWP version with the next release.
It seems that ArtjomP had some experience with it.

**Comment by ArtjomP** [comment]
I'll check it later. Please don't remove the UWP version. I use it a lot. If you need any help with it message me.

**Comment by Aragas** [comment]
Okay, I'll assign you to the issues that had some problems with UWP. If everything there is alright, I'll return it with 1.4.1

**Comment by ArtjomP** [comment]
There is an issue with 1.4.0. 
You added the project for 259 & didn't add the folders implementation this changed behavior .
Sorry I'm a little bit busy right now to investigate the issue in detail.

**Comment by Aragas** [comment]
259 project should be alright, it mirrors the 111 one.

**Comment by ArtjomP** [comment]
I added tests: [commit](https://github.com/ArtjomP/PCLExt.FileStorage/commit/b21b8f89c484c79414a1b165f2c6de53196b3fae) @Aragas if you approve it I'll push a pull request from this commit to the master branch.
I could not reproduce the issue.
What was the version?
At this moment [CheckExistsAsync ](https://github.com/PCLExt/PCLExt.FileStorage/blob/c24f36c6f3359bfc8da3d3e7d02e87ba9b3c5552/src/PCLExt.FileStorage.UWP/StorageFolderImplementation.cs#L99)  catches & it just can't appear.

**Comment by Aragas** [comment]
@ArtjomP sure, looks good. And btw, do we need to check both sync and async versions? In non UWP implementations, async just switches the thread, in UWP, the sync is using the async methods internal

**Comment by ArtjomP** [comment]
Sync version could hang if someone would call it from the Main UI thread it's bad to call sync method in 2K18. We have [AwaitExtensions](https://github.com/PCLExt/PCLExt.FileStorage/blob/master/src/PCLExt.FileStorage.Portable111/Extensions/AwaitExtensions.cs) to switch the main thread. But if we use it everywhere it could be easy misused if someone wants to update UI using async/await
```cs
async void UpdateUi()
{
  // this could throw exception if CheckExistsAsync uses AwaitExtensions->SwitchOffMainThreadAsync
   uiElement.Value = await _folder.CheckExistsAsync("test.zip");
}
```

I propose to leave it as is and solve particular issues if they would appear.

**Comment by Aragas** [comment]
Should be fixed by #32
